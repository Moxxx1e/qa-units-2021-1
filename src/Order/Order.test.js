jest.mock('../utils/getDate');

/* eslint-disable import/first */
import React from 'react'
import {getDate} from "../utils/getDate";
import {shallow, configure} from 'enzyme';
import Order from "./Order";
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Order.js', () => {
    getDate.mockReturnValue(`11 марта, чт, 2021 год`);

    it('render', () => {
        const order = {shop: 'shop', date: 2131212, items: ["some data"]}
        const wrapper = shallow(<Order
            order={order}
        />);
        expect(wrapper).toMatchSnapshot();
        expect(getDate).toHaveBeenCalledTimes(1);
    });

    it('render with empty order', () => {
        const emptyWrapper = shallow(<Order/>);
        expect(emptyWrapper).toMatchSnapshot();
    });

    it('render with null order', () => {
        const order = {shop: null, date: null}
        const nullWraper = shallow(<Order
            order={order}
        />);
        expect(nullWraper).toMatchSnapshot();
    });
})

