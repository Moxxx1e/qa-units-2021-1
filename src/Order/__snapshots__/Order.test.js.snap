// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js render 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    >
      11 марта, чт, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      some data
    </div>
  </div>
</div>
`;

exports[`Order.js render with empty order 1`] = `""`;

exports[`Order.js render with null order 1`] = `""`;
